### PR TITLE
feat: Upgrade wallet to 128-bit cycles

### PR DESCRIPTION
**Pull request by adamspofford-dfinity**
Adds:
- `wallet_balance128`
- `wallet_send128`
- `wallet_create_canister128`
- `wallet_create_wallet128`
- `wallet_call128`
- `get_events128`
- `get_managed_canister_events128`

The stable state is upgraded to 128-bit internally (the 128-bit functions are now the 'real' functions, with the 64-bit functions as wrappers).

### CODE REVIEW

File: wallet/src/lib.rs [hunk 1]
```diff
@@ -776,14 +856,72 @@ struct GetEventsArgs {
...
+                        V1EventKind::AddressAdded { id, name, role }
+                    }
+                    EventKind::AddressRemoved { id } => V1EventKind::AddressRemoved { id },
+                    EventKind::CanisterCalled {
+                        canister,
+                        cycles,
+                        method_name,
+                    } => V1EventKind::CanisterCalled {
+                        canister,
+                        cycles: cycles.try_into().unwrap(),
```
Comment (ericswanson-dfinity) [inline wallet/src/lib.rs:890]: If we used .expect("Event cycles exceeded 64 bit limit.  Call get_events128") instead, would a caller get more information from the resulting panic?

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Managed canisters are now tracked, and the events that pertain to them are tracked under them.
   - Added `list_managed_canisters`, `get_managed_canister_events`, and `set_short_name` functions.
 
+- Each function that deals with a 64-bit cycle count has been paired with a 128-bit equivalent.
```
Comment (ericswanson-dfinity) [inline CHANGELOG.md:19]: Suggestion to call out the following:
- internal data structure now stores 128-bit values
- get_events and get_managed_canister_events will trap if a returned event contains a cycles value that exceeds 64 bits

File: wallet/src/lib.rs (hunk 1 above)
Comment (adamspofford-dfinity) [reply wallet/src/lib.rs:890]: I had expected it to include what the conversion error actually was, but it doesn't, you're right.

File: wallet/src/lib.rs [hunk 3]
```diff
@@ -337,33 +357,48 @@ mod wallet {
...
-        amount: u64,
+        amount: TCycles,
     }
 
     /// Return the cycle balance of this canister.
     #[query(guard = "is_custodian_or_controller", name = "wallet_balance")]
-    fn balance() -> BalanceResult {
+    fn balance() -> BalanceResult<u64> {
         BalanceResult {
             amount: api::canister_balance(),
```
Comment (ericswanson-dfinity) [inline wallet/src/lib.rs:374]: Do you know what the behavior is of `api::canister_balance()` when the balance exceeds 64 bits?  Do you think this should call canister_balance128(), and exit with a similar error message ("call balance128") if so?

Comment (adamspofford-dfinity) [reply wallet/src/lib.rs:374]: It traps with a replica-defined error message. It probably should be defined on the wallet's end, just so the correct function name should be printed.

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**